File: app/src/main/java/com/liarstudio/courierservice/logic/user/UserAPI.java
package com.liarstudio.courierservice.API;

import com.liarstudio.courierservice.entities.user.User;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;


public interface UserAPI {

    //Попытка входа пользователя в систему. При удачном исходе будет возвращен экземпляр класса пользователь
    @GET("/auth")
    Call<User> login(@Query("email") String email, @Query("password") String password);


    //Попытка создания пользователя. При удачном исходе будет возвращен созданный пользователь
    @FormUrlEncoded
    @POST("/auth")
    Call<User> register(
            @Field("email") String email,
            @Field("name") String name,
            @Field("password") String password);



    //Получение списка всех пользователей с указанной в параметрах ролью
    @GET("/users")
    Call<List<User>> loadUsers(@Query("role") int role);


    //Получение пользователя с заданным id
    @GET("/users/{id}")
    Call<User> loadUser(@Path("id") int id);


}

File: app/src/main/java/com/liarstudio/courierservice/logic/user/UserApi.kt
package com.liarstudio.courierservice.logic.user

import com.liarstudio.courierservice.entitiy.user.User
import com.liarstudio.courierservice.logic.UrlUser.GET_USER
import com.liarstudio.courierservice.logic.UrlUser.GET_USERS_ROLE
import io.reactivex.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface UserApi {

    /**
     * Получение списка пользователей
     *
     * @return  Call со списком пользователей
     */
    @GET(GET_USERS_ROLE)
    fun loadUsers(): Observable<List<User>>


    /**
     * Получение пользователя по id
     *
     * @param id идентификатор пользователя
     *
     * @return Call с пользователем
     */
    @GET(GET_USER)
    fun loadUser(@Path("user_id") id: Long): Observable<User>
}